### PR TITLE
[blksync] simplify code in blocksync/buffer

### PR DESCRIPTION
**Pull request by Liuhaai**
-  dequeAll() can return a empty slice instead of `nil`
 - Rename `Delete` to `Pop`

### CODE REVIEW

File: blocksync/buffer.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func newBlockBuffer(bufferSize, intervalSize uint64) *blockBuffer {
 	}
 }
 
-func (b *blockBuffer) Delete(height uint64) []*peerBlock {
+func (b *blockBuffer) Pop(height uint64) []*peerBlock {
```
Comment (CoderZhi) [inline blocksync/buffer.go:38]: why do you rename this function?

**Review by CoderZhi — COMMENTED** [review]
the title is too general. This PR doesn't "refactor" anything, but refine some of the corner cases.

File: blocksync/buffer.go (hunk 1 above)
Comment (Liuhaai) [reply blocksync/buffer.go:38]: `Delete` is to remove buffer of height x, whereas `Pop` is to remove buffer of height x and return the buffer to the caller.

File: blocksync/uniqueue.go [hunk 2]
```diff
@@ -33,9 +33,6 @@ func (uq *uniQueue) enque(blk *peerBlock) {
 }
 
 func (uq *uniQueue) dequeAll() []*peerBlock {
-	if len(uq.blocks) == 0 {
-		return nil
-	}
```
Comment (dustinxie) [inline blocksync/uniqueue.go:38]: i think it's OK to leave this?

**Review by CoderZhi — APPROVED** [review]

File: blocksync/uniqueue.go (hunk 2 above)
Comment (Liuhaai) [reply blocksync/uniqueue.go:38]: It is better to return an empty slice than nil when error type is unavilable in return values

Comment (dustinxie) [reply blocksync/uniqueue.go:38]: ok, then use `return []*peerBlock{}`? 
line 40, 41 is not necessary for the case `if len(uq.blocks) == 0`

**Review by dustinxie — APPROVED** [review]

**Pull request merged**